src/frontend/views/isbn-registry/messages/IsbnMessage.jsx

```diff
@@ -71,6 +70,30 @@ function IsbnMessage(props) {
 
   // Handles back button click
   const handleGoBack = () => {
+    // If coming from publisher registration - redirect to publisher page
+    if (history.location.state.messageCode === 'publisher_registered_isbn'
+      || history.location.state.messageCode === 'publisher_registered_ismn') {
+      return history.push({
+        pathname: `/isbn-registry/publishers/${history.location.state.publisherId}`
+      });
+    }
+
+    // If coming from publication request page - redirect to publication request page
+    if (history.location.state.messageCode === 'identifier_created_isbn'
+      || history.location.state.messageCode === 'identifier_created_ismn') {
+      return history.push({
+        pathname: `/isbn-registry/requests/publications/${history.location.state.publicationId}`
+      });
+    }
+
+    // If coming from batch page - redirect to batch page
+    if (history.location.state.messageCode === 'big_publisher_isbn'
```

Comment:
Proposing to redirect back to publisher view instead of batch view in case where identifier batch information is send.